### PR TITLE
refactor(FX-4736): Use `collector-profile/saves` instead of `collector-profile/saves2` for artwork lists

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Refactor**

This PR solves [FX-4736]

### Description
Review app: https://reuse-save-path.artsy.net/collector-profile/saves

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[FX-4736]: https://artsyproduct.atlassian.net/browse/FX-4736?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Server/middleware/userRequiredMiddleware.ts [hunk 1]
```diff
@@ -9,7 +9,6 @@ const USER_REQUIRED_ROUTES = [
   "/settings/purchases(.*)",
   "/settings/shipping",
   "/user/conversations(.*)",
-  "/collector-profile/saves2(.*)",
```
Comment (dimatretyak) [inline src/Server/middleware/userRequiredMiddleware.ts:12]: Some details are being [discussed](https://artsy.slack.com/archives/C9SATFLUU/p1681497409074369)

File: src/Apps/CollectorProfile/Routes/SavesAndFollows/CollectorProfileSavesAndFollowsRoute.tsx [hunk 2]
```diff
@@ -11,15 +13,25 @@ interface CollectorProfileSavesAndFollowsRouteProps {
...
   {
     me: graphql`
-      fragment CollectorProfileSavesAndFollowsRoute_me on Me {
-        name
+      fragment CollectorProfileSavesAndFollowsRoute_me on Me
+        @argumentDefinitions(
+          shouldFetchArtworkListsData: { type: "Boolean!" }
+        ) {
+        ...CollectorProfileSaves2Route_me
+          @include(if: $shouldFetchArtworkListsData)
```
Comment (dimatretyak) [inline src/Apps/CollectorProfile/Routes/SavesAndFollows/CollectorProfileSavesAndFollowsRoute.tsx:34]: we will fetch this data only if feature flag is enabled

**Comment by dimatretyak** [comment]
**question**: should we add redirect from `collector-profile/saves2` to `collector-profile/saves`? So that all links (for example, in Notion docs) remain working

File: src/Apps/CollectorProfile/Routes/SavesAndFollows/CollectorProfileSavesAndFollowsRoute.tsx (hunk 2 above)
Comment (nickskalkin) [reply src/Apps/CollectorProfile/Routes/SavesAndFollows/CollectorProfileSavesAndFollowsRoute.tsx:34]: Cool, didn't know about this directive together with if!

File: src/Apps/CollectorProfile/Routes/Saves2/Components/__tests__/ArtworkListItem.jest.tsx [hunk 3]
```diff
@@ -83,7 +83,7 @@ describe("ArtworkListItem", () => {
     })
 
     const link = screen.getByRole("link")
-    const href = "/collector-profile/saves2/collection-id"
```
Comment (nickskalkin) [inline src/Apps/CollectorProfile/Routes/Saves2/Components/__tests__/ArtworkListItem.jest.tsx:86]: **suggestion**: let's use BASE_PATH from the [constants file](https://github.com/artsy/force/pull/12255/files#diff-3fae5dc8a2308d9e22e6eed4a0a3763e9e63165f6f46f32054ecd2c0384d6ddd) here?

File: src/Components/NavBar/Menus/NavBarUserMenu.tsx [hunk 4]
```diff
@@ -99,7 +97,7 @@ export const NavBarUserMenu: React.FC = () => {
       >
         <NavBarMenuItemLink
           aria-label="View your Saves"
-          to={savesPath}
+          to="/collector-profile/saves"
```
Comment (nickskalkin) [inline src/Components/NavBar/Menus/NavBarUserMenu.tsx:100]: **suggestion**: let's use BASE_SAVES_PATH here as well?

**Review by nickskalkin — DISMISSED** [review]
👍 Left a couple of tiny suggestions

**Review by dblandin — COMMENTED** [review]
Nice 💯

File: src/Server/middleware/userRequiredMiddleware.ts (hunk 1 above)
Comment (dimatretyak) [reply src/Server/middleware/userRequiredMiddleware.ts:12]: Decided to redirect to login form

**Pull request merged**